var/spack/repos/builtin/packages/pgi/package.py

```diff
@@ -64,6 +64,10 @@ class Pgi(Package):
     license_vars = ['PGROUPD_LICENSE_FILE', 'LM_LICENSE_FILE']
     license_url = 'http://www.pgroup.com/doc/pgiinstall.pdf'
 
+    def url_for_version(self, version):
+        return "file://{0}/pgilinux-20{1}-{2}-x86_64.tar.gz".format(
+            os.getcwd(), version.up_to(1), version.joined)
+
```

Comment:
Not going to hold up this PR for it, but it might make sense to add a `-f` or maybe `--archive` option to `install` so that not every package has to do this.  `-f` would still checksum, but could try to deduce the version from the tarball name, and require a `spec` with a version if it can't figure out the version from the tarball.